src/go/rpk/pkg/cli/topic/mount.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package topic
+
+import (
+	"time"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var alias string
+	var aliasNamespace string
+	var namespace string
+	var location string
+	var timeout int
+
+	cmd := &cobra.Command{
+		Use:   "mount [TOPICS...]",
+		Short: "Mount a topic to the Redpanda Cluster from tiered storage",
+		Long:  helpMount,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			pf, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			config.CheckExitCloudAdmin(pf)
+			adm, err := adminapi.NewClient(fs, pf)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			topic := rpadmin.InboundTopic{
+				SourceTopic: rpadmin.Topic{
+					Topic:     topics[0],
+					Namespace: namespace,
+				},
+			}
+			if alias != "" && aliasNamespace != "" {
+				topic.Alias = &rpadmin.Topic{
+					Topic:     alias,
+					Namespace: aliasNamespace,
+				}
+			}
+			if location != "" {
+				topic.Location = location
+			}
+
+			migration, err := adm.AddInboundMigration(cmd.Context(), rpadmin.InboundMigration{
+				MigrationType:  "inbound",
+				Topics:         []rpadmin.InboundTopic{topic},
+				ConsumerGroups: []string{}, // not implemented yet
+			})
+			out.MaybeDie(err, "unable to add inbound migration: %v", err)
```

Comment:
wdyt of:
```suggestion
			out.MaybeDie(err, "unable to request topic migration: %v", err)
```
An outbound migration means nothing to our end users, this is an implementation detail that we can hide in the err message.